### ISSUE TITLE
Custom quote

### ISSUE THREAD

**Issue by Gerardo-Sista** [nreco/csv#3]
Why there is no possibility to accept a different quote? 

Something like this 
     var csvReader = new CsvReader(stream, delimiter, quote);    (eventually also comment char....)

This lib is very fast but also incomplete with only delimiter. I hope you add this feature.

**Comment by VitaliyMF** [comment]
> This lib is very fast but also incomplete with only delimiter.

you can specify the delimiter instead of comma (like semicolon, tab etc), this is fully supported.

However,  double quotes is an _only_ way how escaped value can be specified in the valid CSV, see https://tools.ietf.org/rfc/rfc4180.txt

**Comment by Gerardo-Sista** [comment]
Thank you for your fast reply,
RFC4180 define double quotes as the only admitted tex qualifier: **we all know this.**
But it's **quite common** that different char is used too: that's why all other libs supports also different text qualifiers.
**I benchmarked various CSV libs and your is the best: fast, easy and no overheads like fields mapping etc.** but I need to parse CSV with custom text qualifier too (not my choice and I cannot change that request). This is NuGet package request too.
Adding enhancement to your lib with custom qualifiers (and eventually comments support) doesn't break your RFC compliancy since default value is always " as defined in RFC specifications.
I hope you'll think about it.

**Comment by VitaliyMF** [comment]
I'm ok to add an option to specify custom character instead of double-quote for escaped values.

Feel free to contribute and submit your PR for this. Don't forget to add this option to both `CsvReader` and `CsvWriter` classes, and add tests for custom quote usage.

**Comment by Gerardo-Sista** [comment]
Thank you very much. 
I will contribute as I can. 
For example, I like CsvHelper flexibility (by easy configuration like field trimming etc too ) but your lib is really incredibly fast. 
Adding the same flexibility will make it the best CSV parser for sure!  :)

**Comment by Gerardo-Sista** [comment]
After arguing with my project manager I decided to stay in full RFC4180 rules because very malformed CSV was given to us.
It's not our responsibility to "try" to solve those issues (if import fails it's not our fault)
**So your initial design point is correct: "this lib is RFC4180 compliant".**

**We will use your lib and force people to provide well formed CSV** with RFC double quotes as text qualifier.

Closing this as solved.

Thank you for your time and your great lib!!
We will be happy to benchmark your further enhancements  when you'll implement them. :)
